### PR TITLE
Allow to retrieve AnnotationReader when entity manager uses MappingChain

### PR DESCRIPTION
**Pull request by danydev**
Adding support retrieving the AnnotationReader when the EntityManager is configured to use a MappingChain containing an AnnotationDriver. Also added an exception that will make more clear what happens if the users uses an EntityManager without AnnotationDriver at all.

NOTE: I did not write tests because the there aren't other tests in the factory. Also, I would recommend introducing Mockery or equivalent to allow test easily classes with dependencies (and this factory is a good candidate) instead of going with integration test providing real objects.

### CODE REVIEW

**Review by PowerKiKi — CHANGES_REQUESTED** [review]
This is an almost duplicate of #2, but it doesn't really matter for me.

We'll definitely have to cover it with tests though. As of now the lib has 100% coverage and we should keep it that way.

What would be your argument in favor of Mockery instead of out-of-the-box PHPUnit ? I am not familiar with the former, but it seems functionally equivalent, so why should we add a new dependency for that ?

**Comment by danydev** [comment]
Oh, I didn't see someone else noticed the same limitation. Yes, the PRs are almost equivalent.

When I said introduce Mockery or equivalent I meant introducing mocks in tests (so if you don't feel comfortable with Mockery's mocks, PHPUnit's mocks are good enough), and that's because from a quick inspection I saw most of the tests are basically integration tests. So mine was just a general advice to make tests more unit-test oriented.

For example adding an integration test for this kind of change, it's a non-obvious thing to achieve, since I would need to deal with Doctrine's EM details configuration. While a unit test approach could make it a bit easier to target it.

I'm going to park the case, but as soon I get some spare time, I'll dig on it to add the test

**Comment by PowerKiKi** [comment]
I agree with you, unit tests would be best. But in this specific case integration tests are not that terrible, because they are super fast and still allows for a very high coverage. But again, I have nothing against unit tests, on the contrary. But I feel integration tests can be faster to write in some cases.

Anyway, I'll wait for your tests to re-evaluate merging this PR. I might prefer going with pure PHPUnit, but if you can show me the benefit of Mockery, then I would be open to it too...

File: src/Factory/AbstractFieldsConfigurationFactory.php [hunk 1]
```diff
@@ -130,7 +132,19 @@ private function isExcluded(ReflectionMethod $method): bool
      */
     protected function getAnnotationReader(): Reader
     {
-        return $this->entityManager->getConfiguration()->getMetadataDriverImpl()->getReader();
+        $mappingDriver = $this->entityManager->getConfiguration()->getMetadataDriverImpl();
+        if ($mappingDriver instanceof AnnotationDriver) {
+            return $mappingDriver->getReader();
+        }
+        if ($mappingDriver instanceof MappingDriverChain) {
+            foreach ($mappingDriver->getDrivers() as $driver) {
```
Comment (TomHAnderson) [inline src/Factory/AbstractFieldsConfigurationFactory.php:140]: This only allows annotations on one driver.  Often when using a MappingDriverChain there will be multiple drivers which may each use annotations.

**Comment by PowerKiKi** [comment]
Duplicated by #19 which looks more promising

**Pull request closed without merging**